src/modules/data/blocks/datetime/sagas.js

```diff
@@ -11,9 +11,66 @@ import {
 	types,
 	selectors,
 	actions,
-	thunks,
 } from '@moderntribe/events/data/blocks/datetime';
-import { date as dateUtil } from '@moderntribe/common/utils';
+import {
+	date as dateUtil,
+	moment as momentUtil,
+	time as timeUtil,
+} from '@moderntribe/common/utils';
+
+const {
+	HALF_HOUR_IN_SECONDS,
+	DAY_IN_SECONDS,
+	HOUR_IN_SECONDS,
+	MINUTE_IN_SECONDS,
+} = timeUtil;
+
+//
+// ──────────────────────────────────────────────────── I ──────────
+//   :::::: D E R I V E : :  :   :    :     :        :          :
+// ──────────────────────────────────────────────────────────────
+//
+
+/**
+ * Convert current start and end date into moments
+ *
+ * @export
+ * @since 4.7.0-beta1
+ * @returns {Object} {start, end}
+ */
+export function* deriveMomentsFromDates() {
+	const dates = yield all( {
+		start: select( selectors.getStart ),
+		end: select( selectors.getEnd ),
+	} );
+
+	return yield all( {
+		start: call( momentUtil.toMoment, dates.start ),
+		end: call( momentUtil.toMoment, dates.end ),
+	} );
+}
+
+/**
+ * Convert current start and end date into seconds
+ *
+ * @export
+ * @since 4.7.0-beta1
+ * @returns {Object} {start, end}
+ */
+export function* deriveSecondsFromDates() {
+	const moments = yield call( deriveMomentsFromDates );
+
+	return yield all( {
+		start: call( timeUtil.toSeconds, moments.start.format( 'HH:mm:ss' ) ),
```

Comment:
There's a moment utility function `toDatabaseTime` which takes in a moment and spits out a `HH:mm:ss` format time. Not sure if you can nest call functions, but you could use that here.